tools/pyboard.py

```diff
@@ -123,7 +123,25 @@ def __init__(self, device, baudrate=115200, user='micro', password='python'):
             self.serial = TelnetToSerial(device, user, password, read_timeout=10)
         else:
             import serial
-            self.serial = serial.Serial(device, baudrate=baudrate, interCharTimeout=1)
+            delayed = False
+            delay = 25
+            for attempt in range(delay):
+                try:
+                    self.serial = serial.Serial(device, baudrate=baudrate, interCharTimeout=1)
+                    done = True
+                    break
+                except serial.serialutil.SerialException:
```

Comment:
serial.serialutil.SerialException is derived from IOError. There are a couple places in the serial code where it translates OSError to SerialException, but apparently not all. In particular, the open call doesn't (it catches IOError but not OSError).